.github/workflows/deploy.yml

```diff
@@ -24,22 +18,11 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-go@v5
         with:
-          go-version: ${{ env.GO_VERSION }}
+          go-version: 1.21
           cache-dependency-path: scripts/zkevm-config-diff/go.sum
 
-      # Install tools.
-      - name: Install kurtosis
-        run: |
-          echo "deb [trusted=yes] https://apt.fury.io/kurtosis-tech/ /" | sudo tee /etc/apt/sources.list.d/kurtosis.list
-          sudo apt update
-          sudo apt install kurtosis-cli=${{ env.KURTOSIS_VERSION }}
-          kurtosis analytics disable
-
-      - name: Install yq
-        run: pip3 install yq
-
-      - name: Install foundry
-        uses: foundry-rs/foundry-toolchain@v1
+      - name: Install Kurtosis CDK tools
+        uses: ./.github/actions/setup-kurtosis-cdk
```

Comment:
nit: here as well, workflow may be more readable and maintainable if we had shared git actions for these tasks